### PR TITLE
Make sure dd selector uses correct keyframe to select packets

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

File: pkg/sfu/buffer/dependencydescriptorparser.go [hunk 1]
```diff
@@ -66,12 +67,15 @@ type ExtDependencyDescriptor struct {
 	ActiveDecodeTargetsUpdated bool
 	Integrity                  bool
 	ExtFrameNum                uint64
+	// the frame number of the keyframe which the current frame depends on
```
Comment (boks1971) [inline pkg/sfu/buffer/dependencydescriptorparser.go:69]: technically, current frame depends on the dependency descriptor structure attached to this key frame, right? not directly the key frame itself. Not a big issue, just trying to understand.

**Review by paulwe — APPROVED** [review]

File: pkg/sfu/buffer/dependencydescriptorparser.go (hunk 1 above)
Comment (boks1971) [reply pkg/sfu/buffer/dependencydescriptorparser.go:69]: Curiosity question: is a structure always attached to a key frame only? Wondering if this should be renamed to `ExtStructureFrameNum` to mean frame number to which this structure was attached.

**Review by boks1971 — APPROVED** [review]
Thank you for digging into this @cnderrauber !!!

Left som minor comments.

File: pkg/sfu/buffer/dependencydescriptorparser.go (hunk 1 above)
Comment (cnderrauber) [reply pkg/sfu/buffer/dependencydescriptorparser.go:69]: Good question, I checked the specification that it doesn't require a structure must be carried by keyframe, but in libwebrtc's frame assembler, it assume that every frame attach a structure is a key frame.

**Pull request merged**